app/tests/conftest.py

```
@@ -0,0 +1,27 @@
+def vault_repo() -> Generator[InMemoryVaultRepository, None, None]:
+    """
+    Fixture that provides an empty in-memory repository
+    and overrides the app's dependency.
+    """
+    repo = InMemoryVaultRepository()
+    app.dependency_overrides[get_vault_repo] = lambda: repo
+    yield repo
+    app.dependency_overrides.clear()
+
+@pytest.fixture
+def client(vault_repo: InMemoryVaultRepository) -> Generator[TestClient, None, None]:
```

Ruff (0.14.14)

[warning] Unused function argument: `vault_repo`

(ARG001)